### PR TITLE
[DROOLS-6627] Verify Scorecard reasoncode when result is null

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @kostola @jiripetrlik

See https://issues.redhat.com/browse/DROOLS-6627

Please note there is an underlying issue not completely fixed, i.e. the "status" of "KiePMMLModel" that it is preserved between multiple invocation.

The overall fix will require a global refactor, and will be addressed by https://issues.redhat.com/browse/DROOLS-6641

This PR must be merged after https://github.com/kiegroup/drools/pull/3868

### CODE REVIEW

**Review by jiripetrlik — APPROVED** [review]

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-drools/kie-pmml-models-drools-common/src/main/java/org/kie/pmml/models/drools/commons/model/KiePMMLDroolsModel.java [hunk 1]
```diff
@@ -75,10 +76,12 @@ public Object evaluate(final Object knowledgeBase, Map<String, Object> requestDa
         String fullClassName = this.getClass().getName();
         String packageName = fullClassName.contains(".") ?
                 fullClassName.substring(0, fullClassName.lastIndexOf('.')) : "";
+        Map<String, Object> holder = new HashMap<>();
         KiePMMLSessionUtils.Builder builder = KiePMMLSessionUtils.builder((KieBase) knowledgeBase, name, packageName,
                                                                           toReturn)
                 .withObjectsInSession(requestData, fieldTypeMap)
-                .withOutputFieldsMap(outputFieldsMap);
+                .withOutputFieldsMap(holder);
+        outputFieldsMap = holder;
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-models/kie-pmml-models-drools/kie-pmml-models-drools-common/src/main/java/org/kie/pmml/models/drools/commons/model/KiePMMLDroolsModel.java]: Is this necessary? I don't usually like a method that changes the instance of a field. Isn't possible to do 
`outpitFieldsMap.putAll(output)`
?
Or do a `clear()` if necessary

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-scorecard/kie-pmml-models-scorecard-model/src/main/java/org/kie/pmml/models/scorecard/model/KiePMMLScorecardModel.java [hunk 2]
```diff
@@ -66,12 +67,15 @@ public Object evaluate(final Object knowledgeBase, final Map<String, Object> req
...
-                                        useReasonCodes,
-                                        baselineScore).orElse(null);
+        Map<String, Object> holder = new HashMap<>();
+        Object toReturn = characteristics.evaluate(defineFunctions, derivedFields, kiePMMLOutputFields, requestData,
+                                                             holder,
+                                                             initialScore,
+                                                             reasonCodeAlgorithm,
+                                                             useReasonCodes,
+                                                             baselineScore).orElse(null);
+        outputFieldsMap = holder;
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-models/kie-pmml-models-scorecard/kie-pmml-models-scorecard-model/src/main/java/org/kie/pmml/models/scorecard/model/KiePMMLScorecardModel.java]: Same here

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-drools/kie-pmml-models-drools-common/src/main/java/org/kie/pmml/models/drools/commons/model/KiePMMLDroolsModel.java (hunk 1 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-models/kie-pmml-models-drools/kie-pmml-models-drools-common/src/main/java/org/kie/pmml/models/drools/commons/model/KiePMMLDroolsModel.java]: @danielezonca 
done

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-scorecard/kie-pmml-models-scorecard-model/src/main/java/org/kie/pmml/models/scorecard/model/KiePMMLScorecardModel.java (hunk 2 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-models/kie-pmml-models-scorecard/kie-pmml-models-scorecard-model/src/main/java/org/kie/pmml/models/scorecard/model/KiePMMLScorecardModel.java]: @danielezonca 
done

**Review by danielezonca — APPROVED** [review]

**Pull request merged**